File: auto-generated/php/SwaggerClient-php/lib/Model/TradingVolume.php
<?php
/**
 * TradingVolume
 *
 * PHP version 5
 *
 * @category Class
 * @package  Swagger\Client
 * @author   Swagger Codegen team
 * @link     https://github.com/swagger-api/swagger-codegen
 */

/**
 * BitMEX API
 *
 * ## REST API for the BitMEX Trading Platform  _If you are building automated tools, please subscribe to the_ _[BitMEX API RSS Feed](https://blog.bitmex.com/api_announcement/feed/) for changes. The feed will be updated_ _regularly and is the most reliable way to get downtime and update announcements._  [View Changelog](/app/apiChangelog)  -  #### Getting Started  Base URI: [https://www.bitmex.com/api/v1](/api/v1)  ##### Fetching Data  All REST endpoints are documented below. You can try out any query right from this interface.  Most table queries accept `count`, `start`, and `reverse` params. Set `reverse=true` to get rows newest-first.  Additional documentation regarding filters, timestamps, and authentication is available in [the main API documentation](/app/restAPI).  _All_ table data is available via the [Websocket](/app/wsAPI). We highly recommend using the socket if you want to have the quickest possible data without being subject to ratelimits.  ##### Return Types  By default, all data is returned as JSON. Send `?_format=csv` to get CSV data or `?_format=xml` to get XML data.  ##### Trade Data Queries  _This is only a small subset of what is available, to get you started._  Fill in the parameters and click the `Try it out!` button to try any of these queries.  - [Pricing Data](#!/Quote/Quote_get)  - [Trade Data](#!/Trade/Trade_get)  - [OrderBook Data](#!/OrderBook/OrderBook_getL2)  - [Settlement Data](#!/Settlement/Settlement_get)  - [Exchange Statistics](#!/Stats/Stats_history)  Every function of the BitMEX.com platform is exposed here and documented. Many more functions are available.  ##### Swagger Specification  [⇩ Download Swagger JSON](swagger.json)  -  ## All API Endpoints  Click to expand a section.
 *
 * OpenAPI spec version: 1.2.0
 * Contact: support@bitmex.com
 * Generated by: https://github.com/swagger-api/swagger-codegen.git
 * Swagger Codegen version: 2.4.42-SNAPSHOT
 */

/**
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen
 * Do not edit the class manually.
 */

namespace Swagger\Client\Model;

use \ArrayAccess;
use \Swagger\Client\ObjectSerializer;

/**
 * TradingVolume Class Doc Comment
 *
 * @category Class
 * @description 30 days USD average trading volume
 * @package  Swagger\Client
 * @author   Swagger Codegen team
 * @link     https://github.com/swagger-api/swagger-codegen
 */
class TradingVolume implements ModelInterface, ArrayAccess
{
    const DISCRIMINATOR = null;

    /**
      * The original name of the model.
      *
      * @var string
      */
    protected static $swaggerModelName = 'TradingVolume';

    /**
      * Array of property to type mappings. Used for (de)serialization
      *
      * @var string[]
      */
    protected static $swaggerTypes = [
        'adv_usd' => 'double',
        'adv_usd_spot' => 'double',
        'adv_usd_contract' => 'double'
    ];

    /**
      * Array of property to format mappings. Used for (de)serialization
      *
      * @var string[]
      */
    protected static $swaggerFormats = [
        'adv_usd' => 'double',
        'adv_usd_spot' => 'double',
        'adv_usd_contract' => 'double'
    ];

    /**
     * Array of property to type mappings. Used for (de)serialization
     *
     * @return array
     */
    public static function swaggerTypes()
    {
        return self::$swaggerTypes;
    }

    /**
     * Array of property to format mappings. Used for (de)serialization
     *
     * @return array
     */
    public static function swaggerFormats()
    {
        return self::$swaggerFormats;
    }

    /**
     * Array of attributes where the key is the local name,
     * and the value is the original name
     *
     * @var string[]
     */
    protected static $attributeMap = [
        'adv_usd' => 'advUsd',
        'adv_usd_spot' => 'advUsdSpot',
        'adv_usd_contract' => 'advUsdContract'
    ];

    /**
     * Array of attributes to setter functions (for deserialization of responses)
     *
     * @var string[]
     */
    protected static $setters = [
        'adv_usd' => 'setAdvUsd',
        'adv_usd_spot' => 'setAdvUsdSpot',
        'adv_usd_contract' => 'setAdvUsdContract'
    ];

    /**
     * Array of attributes to getter functions (for serialization of requests)
     *
     * @var string[]
     */
    protected static $getters = [
        'adv_usd' => 'getAdvUsd',
        'adv_usd_spot' => 'getAdvUsdSpot',
        'adv_usd_contract' => 'getAdvUsdContract'
    ];

    /**
     * Array of attributes where the key is the local name,
     * and the value is the original name
     *
     * @return array
     */
    public static function attributeMap()
    {
        return self::$attributeMap;
    }

    /**
     * Array of attributes to setter functions (for deserialization of responses)
     *
     * @return array
     */
    public static function setters()
    {
        return self::$setters;
    }

    /**
     * Array of attributes to getter functions (for serialization of requests)
     *
     * @return array
     */
    public static function getters()
    {
        return self::$getters;
    }

    /**
     * The original name of the model.
     *
     * @return string
     */
    public function getModelName()
    {
        return self::$swaggerModelName;
    }

    

    

    /**
     * Associative array for storing property values
     *
     * @var mixed[]
     */
    protected $container = [];

    /**
     * Constructor
     *
     * @param mixed[] $data Associated array of property values
     *                      initializing the model
     */
    public function __construct(array $data = null)
    {
        $this->container['adv_usd'] = isset($data['adv_usd']) ? $data['adv_usd'] : null;
        $this->container['adv_usd_spot'] = isset($data['adv_usd_spot']) ? $data['adv_usd_spot'] : null;
        $this->container['adv_usd_contract'] = isset($data['adv_usd_contract']) ? $data['adv_usd_contract'] : null;
    }

    /**
     * Show all the invalid properties with reasons.
     *
     * @return array invalid properties with reasons
     */
    public function listInvalidProperties()
    {
        $invalidProperties = [];

        if ($this->container['adv_usd'] === null) {
            $invalidProperties[] = "'adv_usd' can't be null";
        }
        return $invalidProperties;
    }

    /**
     * Validate all the properties in the model
     * return true if all passed
     *
     * @return bool True if all properties are valid
     */
    public function valid()
    {
        return count($this->listInvalidProperties()) === 0;
    }


    /**
     * Gets adv_usd
     *
     * @return double
     */
    public function getAdvUsd()
    {
        return $this->container['adv_usd'];
    }

    /**
     * Sets adv_usd
     *
     * @param double $adv_usd adv_usd
     *
     * @return $this
     */
    public function setAdvUsd($adv_usd)
    {
        $this->container['adv_usd'] = $adv_usd;

        return $this;
    }

    /**
     * Gets adv_usd_spot
     *
     * @return double
     */
    public function getAdvUsdSpot()
    {
        return $this->container['adv_usd_spot'];
    }

    /**
     * Sets adv_usd_spot
     *
     * @param double $adv_usd_spot adv_usd_spot
     *
     * @return $this
     */
    public function setAdvUsdSpot($adv_usd_spot)
    {
        $this->container['adv_usd_spot'] = $adv_usd_spot;

        return $this;
    }

    /**
     * Gets adv_usd_contract
     *
     * @return double
     */
    public function getAdvUsdContract()
    {
        return $this->container['adv_usd_contract'];
    }

    /**
     * Sets adv_usd_contract
     *
     * @param double $adv_usd_contract adv_usd_contract
     *
     * @return $this
     */
    public function setAdvUsdContract($adv_usd_contract)
    {
        $this->container['adv_usd_contract'] = $adv_usd_contract;

        return $this;
    }
    /**
     * Returns true if offset exists. False otherwise.
     *
     * @param integer $offset Offset
     *
     * @return boolean
     */
    #[\ReturnTypeWillChange]
    public function offsetExists($offset)
    {
        return isset($this->container[$offset]);
    }

    /**
     * Gets offset.
     *
     * @param integer $offset Offset
     *
     * @return mixed
     */
    #[\ReturnTypeWillChange]
    public function offsetGet($offset)
    {
        return isset($this->container[$offset]) ? $this->container[$offset] : null;
    }

    /**
     * Sets value based on offset.
     *
     * @param integer $offset Offset
     * @param mixed   $value  Value to be set
     *
     * @return void
     */
    #[\ReturnTypeWillChange]
    public function offsetSet($offset, $value)
    {
        if (is_null($offset)) {
            $this->container[] = $value;
        } else {
            $this->container[$offset] = $value;
        }
    }

    /**
     * Unsets offset.
     *
     * @param integer $offset Offset
     *
     * @return void
     */
    #[\ReturnTypeWillChange]
    public function offsetUnset($offset)
    {
        unset($this->container[$offset]);
    }

    /**
     * Gets the string presentation of the object
     *
     * @return string
     */
    public function __toString()
    {
        if (defined('JSON_PRETTY_PRINT')) { // use JSON pretty print
            return json_encode(
                ObjectSerializer::sanitizeForSerialization($this),
                JSON_PRETTY_PRINT
            );
        }

        return json_encode(ObjectSerializer::sanitizeForSerialization($this));
    }
}

File: auto-generated/cpprest/model/TradingVolume.cpp
/**
 * BitMEX API
 * ## REST API for the BitMEX Trading Platform  _If you are building automated tools, please subscribe to the_ _[BitMEX API RSS Feed](https://blog.bitmex.com/api_announcement/feed/) for changes. The feed will be updated_ _regularly and is the most reliable way to get downtime and update announcements._  [View Changelog](/app/apiChangelog)  -  #### Getting Started  Base URI: [https://www.bitmex.com/api/v1](/api/v1)  ##### Fetching Data  All REST endpoints are documented below. You can try out any query right from this interface.  Most table queries accept `count`, `start`, and `reverse` params. Set `reverse=true` to get rows newest-first.  Additional documentation regarding filters, timestamps, and authentication is available in [the main API documentation](/app/restAPI).  _All_ table data is available via the [Websocket](/app/wsAPI). We highly recommend using the socket if you want to have the quickest possible data without being subject to ratelimits.  ##### Return Types  By default, all data is returned as JSON. Send `?_format=csv` to get CSV data or `?_format=xml` to get XML data.  ##### Trade Data Queries  _This is only a small subset of what is available, to get you started._  Fill in the parameters and click the `Try it out!` button to try any of these queries.  - [Pricing Data](#!/Quote/Quote_get)  - [Trade Data](#!/Trade/Trade_get)  - [OrderBook Data](#!/OrderBook/OrderBook_getL2)  - [Settlement Data](#!/Settlement/Settlement_get)  - [Exchange Statistics](#!/Stats/Stats_history)  Every function of the BitMEX.com platform is exposed here and documented. Many more functions are available.  ##### Swagger Specification  [⇩ Download Swagger JSON](swagger.json)  -  ## All API Endpoints  Click to expand a section. 
 *
 * OpenAPI spec version: 1.2.0
 * Contact: support@bitmex.com
 *
 * NOTE: This class is auto generated by the swagger code generator 2.4.42-SNAPSHOT.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 */



#include "TradingVolume.h"

namespace io {
namespace swagger {
namespace client {
namespace model {

TradingVolume::TradingVolume()
{
    m_AdvUsd = 0.0;
    m_AdvUsdSpot = 0.0;
    m_AdvUsdSpotIsSet = false;
    m_AdvUsdContract = 0.0;
    m_AdvUsdContractIsSet = false;
}

TradingVolume::~TradingVolume()
{
}

void TradingVolume::validate()
{
    // TODO: implement validation
}

web::json::value TradingVolume::toJson() const
{
    web::json::value val = web::json::value::object();

    val[utility::conversions::to_string_t("advUsd")] = ModelBase::toJson(m_AdvUsd);
    if(m_AdvUsdSpotIsSet)
    {
        val[utility::conversions::to_string_t("advUsdSpot")] = ModelBase::toJson(m_AdvUsdSpot);
    }
    if(m_AdvUsdContractIsSet)
    {
        val[utility::conversions::to_string_t("advUsdContract")] = ModelBase::toJson(m_AdvUsdContract);
    }

    return val;
}

void TradingVolume::fromJson(web::json::value& val)
{
    setAdvUsd(ModelBase::doubleFromJson(val[utility::conversions::to_string_t("advUsd")]));
    if(val.has_field(utility::conversions::to_string_t("advUsdSpot")))
    {
        web::json::value& fieldValue = val[utility::conversions::to_string_t("advUsdSpot")];
        if(!fieldValue.is_null())
        {
            setAdvUsdSpot(ModelBase::doubleFromJson(fieldValue));
        }
    }
    if(val.has_field(utility::conversions::to_string_t("advUsdContract")))
    {
        web::json::value& fieldValue = val[utility::conversions::to_string_t("advUsdContract")];
        if(!fieldValue.is_null())
        {
            setAdvUsdContract(ModelBase::doubleFromJson(fieldValue));
        }
    }
}

void TradingVolume::toMultipart(std::shared_ptr<MultipartFormData> multipart, const utility::string_t& prefix) const
{
    utility::string_t namePrefix = prefix;
    if(namePrefix.size() > 0 && namePrefix.substr(namePrefix.size() - 1) != utility::conversions::to_string_t("."))
    {
        namePrefix += utility::conversions::to_string_t(".");
    }

    multipart->add(ModelBase::toHttpContent(namePrefix + utility::conversions::to_string_t("advUsd"), m_AdvUsd));
    if(m_AdvUsdSpotIsSet)
    {
        multipart->add(ModelBase::toHttpContent(namePrefix + utility::conversions::to_string_t("advUsdSpot"), m_AdvUsdSpot));
    }
    if(m_AdvUsdContractIsSet)
    {
        multipart->add(ModelBase::toHttpContent(namePrefix + utility::conversions::to_string_t("advUsdContract"), m_AdvUsdContract));
    }
}

void TradingVolume::fromMultiPart(std::shared_ptr<MultipartFormData> multipart, const utility::string_t& prefix)
{
    utility::string_t namePrefix = prefix;
    if(namePrefix.size() > 0 && namePrefix.substr(namePrefix.size() - 1) != utility::conversions::to_string_t("."))
    {
        namePrefix += utility::conversions::to_string_t(".");
    }

    setAdvUsd(ModelBase::doubleFromHttpContent(multipart->getContent(utility::conversions::to_string_t("advUsd"))));
    if(multipart->hasContent(utility::conversions::to_string_t("advUsdSpot")))
    {
        setAdvUsdSpot(ModelBase::doubleFromHttpContent(multipart->getContent(utility::conversions::to_string_t("advUsdSpot"))));
    }
    if(multipart->hasContent(utility::conversions::to_string_t("advUsdContract")))
    {
        setAdvUsdContract(ModelBase::doubleFromHttpContent(multipart->getContent(utility::conversions::to_string_t("advUsdContract"))));
    }
}

double TradingVolume::getAdvUsd() const
{
    return m_AdvUsd;
}


void TradingVolume::setAdvUsd(double value)
{
    m_AdvUsd = value;
    
}
double TradingVolume::getAdvUsdSpot() const
{
    return m_AdvUsdSpot;
}


void TradingVolume::setAdvUsdSpot(double value)
{
    m_AdvUsdSpot = value;
    m_AdvUsdSpotIsSet = true;
}
bool TradingVolume::advUsdSpotIsSet() const
{
    return m_AdvUsdSpotIsSet;
}

void TradingVolume::unsetAdvUsdSpot()
{
    m_AdvUsdSpotIsSet = false;
}

double TradingVolume::getAdvUsdContract() const
{
    return m_AdvUsdContract;
}


void TradingVolume::setAdvUsdContract(double value)
{
    m_AdvUsdContract = value;
    m_AdvUsdContractIsSet = true;
}
bool TradingVolume::advUsdContractIsSet() const
{
    return m_AdvUsdContractIsSet;
}

void TradingVolume::unsetAdvUsdContract()
{
    m_AdvUsdContractIsSet = false;
}

}
}
}
}

